ansible/roles/test/tasks/snmp/phys_table.yml

```diff
@@ -0,0 +1,54 @@
+# Gather facts with SNMP version 2
+- name: Gathering basic snmp facts about the device
+  snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+  connection: local
+
+- set_fact:
+    chassis_id: "1"
+    sensor_partial_oids: {
+      "Temperature": 1,
+      "Voltage": 2,
+      "Rx1Power": 11,
+      "Rx2Power": 21,
+      "Rx3Power": 31,
+      "Rx4Power": 41,
+      "Tx1Bias": 12,
+      "Tx2Bias": 22,
+      "Tx3Bias": 32,
+      "Tx4Bias": 42,
+      "Tx1Power": 13,
+      "Tx2Power": 23,
+      "Tx3Power": 33,
+      "Tx4Power": 43
+    }
+
+- name: Check chassis entity exists in snmp
+  fail: msg="No chassis entry in Physical Table MIB"
+  when: "{{ chassis_id not in snmp_physical_entities or snmp_physical_entities[chassis_id].entPhysClass != 3 }}"
+
+- debug:
+    msg: "{{ snmp_physical_entities.keys()  }}"
+
+- name: Check all transceivers present in Entity MIB
+  fail:
+    msg: '{{ (((item.value.name.split("Ethernet")[-1] | int + 1) * 1000) | string) }}'
```

Comment:
Could you explain the complex expression?

Reply:
@qiluo-msft the OID index is defined as ifindex*1000, and for EthernetN, ifindex is (N+1), so the OID index for EthernetN will be (N +1)*1000,  detail info can be found in the requirement https://github.com/Azure/SONiC/blob/gh-pages/doc/OIDsforSensorandTransciver.MD#transceiver-requirements-entity-mib

Reply:
The comment is perfect for code comment. Every future code reader will have the same problem

Reply:
@qiluo-msft will add this to code as comments.